nautobot/core/api/serializers.py

```diff
@@ -122,13 +123,50 @@ class BaseModelSerializer(OptInFieldsMixin, serializers.HyperlinkedModelSerializ
     display = serializers.SerializerMethodField(read_only=True, help_text="Human friendly display value")
     object_type = ObjectTypeField()
     composite_key = serializers.SerializerMethodField()
+    natural_keys_values = []
+    related_model_fields_names = []
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         # If it is not a Nested Serializer, we should set the depth argument to whatever is in the request's context
         if not self.is_nested:
             self.Meta.depth = self.context.get("depth", 0)
 
+        # Check if the request is related to CSV export;
+        if self._is_csv_request() and self.instance:
+            related_fields_natural_key_field_lookups = self._get_related_fields_natural_key_field_lookups()
+            if isinstance(self.instance, QuerySet):
```

Comment:
When is this the case?

Reply:
When `many=True` i.e `DeviceSerializer(queryset, many=True)`